src/Views/ListView/Lists/MusicListView.vala

```diff
@@ -195,15 +148,52 @@ public class Noise.MusicListView : GenericList {
         update_sensitivities ();
     }
 
-    public void popup_media_menu (Gee.Collection<Media> selection) {
-        // Create add-to-playlist menu
-        var addToPlaylistMenu = new Gtk.Menu ();
+    public override void update_sensitivities () {
+        media_action_menu.show_all ();
+        
+        switch (hint) {
+            case ViewWrapper.Hint.MUSIC:
+                media_remove.label = _("Remove from Library");
+                import_to_library.visible = false;
+                break;
+            case ViewWrapper.Hint.PLAYLIST:
+                import_to_library.visible = false;
+                break;
+            case ViewWrapper.Hint.READ_ONLY_PLAYLIST:
+                import_to_library.visible = false;
+                if (playlist == App.player.queue_playlist) {
+                    media_remove.label = _("Remove from Queue");
+                    media_menu_queue.visible = false;
+                } else {
+                    media_remove.visible = false;
+                }
+                break;
+            case ViewWrapper.Hint.SMART_PLAYLIST:
```

Comment:
This is the same as the default case, so you can just group it down there by `default:`

Reply:
group it or remove the case?

Reply:
I guess it could just be removed :)